src/context.c

```diff
@@ -1645,8 +1645,7 @@ getdns_context_set_namespaces(getdns_context *context,
 	size_t i;
 	getdns_return_t r = GETDNS_RETURN_GOOD;
 
-	if (!context)
-		return GETDNS_RETURN_INVALID_PARAMETER;
+	RETURN_IF_NULL(context, GETDNS_RETURN_INVALID_PARAMETER);
```

Comment:
I was just trying to be more consistent with the other existing functions - the macro was already there and was used in many other set() functions. I don't mind which one we use, but I like code consistency. 

I also have a question about the return codes. In inspecting all the existing setters, the GETDNS_RETURN_INVALID_PARAMETER is used if the context is NULL and as far as I can see GETDNS_RETURN_CONTEXT_UPDATE_FAIL is used for all other cases, even when a passed parameter is invalid (null pointer to list, invalid enum).  This seems a bit lacking, but I don't know if this was a design decision way back?